### PR TITLE
Improve UX of msg generation process

### LINKED ISSUE #750: Improve message generation experience
To help improve the experience of developers working with rclnodejs I would like to:
1. Make it more obvious to users that rclnodejs generates corresponding javascript classes for each ROS2 interface (msg, srv, action) and that it is the user's responsibility to regenerate these js files when new or revised ROS2 interfaces are added to his system.
2. Improve the msg generation process

For #1a, I would like to add a section to the readme file and/or a wiki page that exposes users to the fact that msg files are being generated and it is the user's responsibility to regenerate msgs when ros2 interfaces change.
#1b Consider including more info in the load_interfaces.js class error msgs when a msg file is not found. Currently rclnodejs throws an exception like this ``The message required does not exist: ${packageName}, ${type}, ${messageName}``. Perhaps the error msg could provide info about how to fix the problem by regenerating msg files?

For #2, I propose we define generate_messages.js as a bin script, e.g.
```
// package.json
"scripts": {
  "generate-messages": "node ./scripts/generate_messages.js"
},
"bin": {
    "generate-ros-messages": "./scripts/generate_messages.js"
},

// generate_messages.js
#!/usr/bin/env node
...

Then a user could run
```
npm node_modules/.bin/generate-ros-messages
```
or create a package script such as:
```
"scripts": {
   "genmsgs": "generate-ros-messages"
}

Looking for thoughts on the proposed action items and additional ideas for improving the user experience of working with msg files.

A final thought is to separate message generation to it's own node package that could be installed as a global package and run directly from commandline, e.g. `generate-ros-messages`.

### PR DESCRIPTION
**Pull request by wayneparrott**
Improve the rclnodejs message generation experience.

1. Configure script/generate_messages.js as a nodejs commandline script names generate-ros-messages

2. Added new README section ROS2 Interface Message Generation. Included are instructions on adding an npm script named generate-ros-messages to an app package that includes rclnodejs as a dependent.

TODO: No tests have been implemented yet. This is a WIP commit for early review.

Fix #750

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,14 +89,54 @@ rclnodejs.init().then(() => {
...
+````
+
+To run the script use `npm` as follows:
+```
+npm run generate-ros-messages
+```
+The newly generated JavaScript files can be found at 
+`<yourproject>/node_modules/rclnodejs/generated/`.
+
+
```
Comment (minggangw) [inline README.md:148]: Could we split the current README into several docs and add the link into README because it's pretty long now :)

Comment (wayneparrott) [reply README.md:148]: Agree! It's got too much going on. How do you envision restructuring it? Actually, that discussion is probably better deferred to a new issue on the topic.

Comment (minggangw) [reply README.md:148]: I'd like to keep the key information in the README and put other sections to other `.md`. How about moving the following sections into separate ones:

- Build from Scratch
- Troubleshooting
- Contributors (CONTRIBUTORS.md?)

I believe you are better than me in organizing the structures and make it more reasonable linguistically as a native speaker :smile:

Comment (wayneparrott) [reply README.md:148]: Agree. I'll start refactoring in the next 1 - 2 days. What for follow up questions about specific requirements or ideas that may come up in the process.

Comment (minggangw) [reply README.md:148]: Thanks, please take your time and enjoy the Christmas :christmas_tree:

**Comment by wayneparrott** [comment]
I reduced this PR down to the following:
1) define a npm binary script in package.json named `generate-messages`
2) add #! details to generate_messages.js script
3) add section to FAQ.md describing how to generate messages
4) added test case that runs the generate-messages script from root folder of a new node pkg, <my_pkg>/node_modules/.bin/generate-messages

It looks like there is a problem with my new test-generate-messages-bin running in CI that I'll track down asap.

**Comment by minggangw** [comment]
Hi @wayneparrott is the PR ready now? because I noticed that there are some conflicts.

**Comment by wayneparrott** [comment]
@minggangw Work has been a bit busy lately. I'm not sure what's up with CI but test-raw-pub-sub.js is failing on mac. The tests I added for this PR pass consistently on all platforms.

**Review by minggangw — COMMENTED** [review]
Please rebase the latest `develop` branch and the failure on macOS will be gone.

**Comment by wayneparrott** [comment]
I'm good to go on this PR.

File: README.md [hunk 2]
```diff
@@ -5,7 +5,14 @@
 | develop | [![Build Status](https://travis-ci.org/RobotWebTools/rclnodejs.svg?branch=develop)](https://travis-ci.org/RobotWebTools/rclnodejs) | [![macOS Build Status](https://circleci.com/gh/RobotWebTools/rclnodejs/tree/develop.svg?style=shield)](https://circleci.com/gh/RobotWebTools/rclnodejs) | [![Build status](https://ci.appveyor.com/api/projects/status/upbc7tavdag1aa5e/branch/develop?svg=true)](https://ci.appveyor.com/project/minggangw/rclnodejs/branch/develop) |
 | master  | [![Build Status](https://travis-ci.org/RobotWebTools/rclnodejs.svg?branch=master)](https://travis-ci.org/RobotWebTools/rclnodejs)  | [![macOS Build Status](https://circleci.com/gh/RobotWebTools/rclnodejs/tree/master.svg?style=shield)](https://circleci.com/gh/RobotWebTools/rclnodejs)  |  [![Build status](https://ci.appveyor.com/api/projects/status/upbc7tavdag1aa5e/branch/master?svg=true)](https://ci.appveyor.com/project/minggangw/rclnodejs/branch/master)  |
 
+<<<<<<< Upstream, based on develop
```
Comment (minggangw) [inline README.md]: Please resolve the conflict

File: README.md [hunk 3]
```diff
@@ -67,11 +74,15 @@ npm i rclnodejs@x.y.z
 
 API documentation is generated by `jsdoc` and can be viewed in the `docs/` folder or [on-line](http://robotwebtools.org/rclnodejs/docs/index.html). To create a local copy of the documentation run `npm run docs`.
 
-## Using TypeScript
+## Using rclnodejs with TypeScript
 
 `rclnodejs` API can be used in TypeScript projects. You can find the TypeScript declaration files (\*.d.ts) in the `types/` folder.
 
+<<<<<<< Upstream, based on develop
```
Comment (minggangw) [inline README.md]: ditto

File: README.md [hunk 4]
```diff
@@ -96,13 +107,57 @@ rclnodejs.init().then(() => {
 });
 ```
 
+<<<<<<< Upstream, based on develop
```
Comment (minggangw) [inline README.md]: ditto

**Review by minggangw — APPROVED** [review]
LGTM, please address the conflicts before merging, thanks!

**Pull request merged**